src/common/contexts/provider.tsx

```diff
@@ -49,14 +50,11 @@ export const ProviderContext = createContext<ProviderContextProps>({
   changeNetwork: async (_chainId: ChainId) => {},
   // eslint-disable-next-line @typescript-eslint/no-empty-function
   reloadNetwork: async () => {},
-  // eslint-disable-next-line @typescript-eslint/no-empty-function
-  getTransactor: () => undefined,
-  // eslint-disable-next-line @typescript-eslint/no-empty-function
-  getProvider: () => undefined,
-  // eslint-disable-next-line @typescript-eslint/no-empty-function
-  getSigner: () => undefined,
   supportedChainInfoObjects: [],
   currentChainId: undefined,
+  provider: currentProvider,
+  providerOrSigner: currentProvider,
```

Comment:
qn: providerOrSigner should it be separated?

Reply:
The contract from typechain takes in a provider or signer object. Without this, everywhere else that uses the provider will need to check between provider and signer and whether wallet is connected.

Reply:
ic, able to point me to the typechain part?

Reply:
For eg, you can see in here, https://github.com/TradeTrust/tradetrust-website/blob/master/src/common/hooks/useTitleEscrowContract.ts#L14, it takes in provider or signer so that in here https://github.com/TradeTrust/tradetrust-website/blob/master/src/common/hooks/useTitleEscrowContract.ts#L25 it can be used in the second parameter.